devlab/tests/generate_load_real_env.py

```diff
@@ -17,83 +86,244 @@ def create_keypairs(self):
         try:
             for user, keypair in zip(self.config.users, self.config.keypairs):
                 if user['enabled']:
-                    self.switch_user(user=user['name'], tenant=user['tenant'],
+                    self.switch_user(user=user['name'],
+                                     tenant=self.update_name_with_prefix
+                                     (user['tenant']),
                                      password=user['password'])
+                    keypair['name'] = self.prefix + keypair['name']
                     self.novaclient.keypairs.create(**keypair)
+                    logging.info('>>> keypair "%s" created for user "%s"',
+                                 keypair['name'], user['name'])
             self.switch_user(user=self.username, password=self.password,
                              tenant=self.tenant)
-        except Exception as e:
-            print "Keypair failed to create:\n %s" % (repr(e))
+        except nv_exceptions.ClientException as e:
+            logging.error('>>>Keypair failed to create:\n "%s"', (repr(e)))
+
+    def create_tenants(self, tenants=None):
+        if tenants is None:
+            tenants = self.config.tenants
+        for tenant in tenants:
+            tenant['name'] = self.prefix + tenant['name']
+            for tnt in self.keystoneclient.tenants.list():
+                if tnt.name == tenant['name']:
+                    print ">>>Tenant", tenant['name'],
+                    print "and its resources already exist."
+                    print ">>>Please run --clean to clean tenant resources"
+                    logging.info(">>>Tenant %s & its resources already exist",
+                                 tenant['name'])
+                    logging.info(">>>Please run --clean first to"
+                                 " clean existing tenant resources")
```

Comment:
mixture of `logging` and `print`. Please leave only one of the types of output, I'd suggest leaving `logging`.